### PR TITLE
standardize docs to `with DAG`

### PR DESCRIPTION
**Pull request by fritz-astronomer**
Switch the `core-concepts/dags.html` page to use `with DAG` except where it makes explicit sense to use `dag1 = DAG(...)`

### CODE REVIEW

File: docs/apache-airflow/core-concepts/dags.rst [hunk 1]
```diff
@@ -648,20 +652,20 @@ This is especially useful if your tasks are built dynamically from configuration
     ### My great DAG
     """
     import pendulum
+    from airflow.operators.empty import EmptyOperator
```
Comment (fritz-astronomer) [inline docs/apache-airflow/core-concepts/dags.rst:653]: additionally - made this snippet more fleshed out

File: docs/apache-airflow/core-concepts/dags.rst [hunk 2]
```diff
@@ -148,10 +148,14 @@ This means you can define multiple DAGs per Python file, or even spread one very
 
 Note, though, that when Airflow comes to load DAGs from a Python file, it will only pull any objects at the *top level* that are a DAG instance. For example, take this DAG file::
 
-    dag_1 = DAG('this_dag_will_be_discovered')
+    with DAG('this_dag_will_be_discovered') as dag:
```
Comment (jedcunningham) [inline docs/apache-airflow/core-concepts/dags.rst:151]: This whole section is wrong with `auto_register` now :(

Can you refactor this whole section?

File: docs/apache-airflow/core-concepts/dags.rst [hunk 3]
```diff
@@ -448,32 +452,32 @@ You can also combine this with the :ref:`concepts:depends-on-past` functionality
         from airflow.models import DAG
         from airflow.operators.empty import EmptyOperator
 
-        dag = DAG(
+        with DAG(
             dag_id="branch_without_trigger",
             schedule="@once",
             start_date=pendulum.datetime(2019, 2, 28, tz="UTC"),
-        )
+        ) as dag:
```
Comment (jedcunningham) [inline docs/apache-airflow/core-concepts/dags.rst]:
```suggestion
        ):
```

Don't actually need this any longer :)

File: docs/apache-airflow/core-concepts/dags.rst [hunk 4]
```diff
@@ -648,20 +652,20 @@ This is especially useful if your tasks are built dynamically from configuration
...
-    )
-    dag.doc_md = __doc__
-
-    t = BashOperator("foo", dag=dag)
-    t.doc_md = """\
-    #Title"
-    Here's a [url](www.airbnb.com)
-    """
+        doc_md=__doc__
+    ) as dag:
```
Comment (jedcunningham) [inline docs/apache-airflow/core-concepts/dags.rst]:
```suggestion
    ):
```

File: docs/apache-airflow/core-concepts/dags.rst (hunk 2 above)
Comment (josh-fell) [reply docs/apache-airflow/core-concepts/dags.rst:151]: @fritz-astronomer Any chance these `as dag:` bits could be removed as @jedcunningham mentioned?

**Pull request closed without merging**